### ISSUE TITLE
[riverpod_lint] - A lint plugin threw an exceptiondart(custom_lint_get_lint_fail)

### ISSUE THREAD

**Issue by emirchus** [rrousselGit/riverpod#2173]
**Describe the bug**
Just when I install it, throws an exception

**To Reproduce**
Install and configure the analyzer

pubspec.yaml:
```yaml
dev_dependencies:
  build_runner: ^2.3.2
  flutter_lints: ^2.0.1
  flutter_test:
    sdk: flutter
  hive_generator: ^2.0.0
  mockito: ^5.3.2
  injectable_generator: ^2.1.3
  custom_lint:
  riverpod_lint:
```

analysis_options.yaml:
````yaml
include: package:flutter_lints/flutter.yaml

linter:
  rules:

analyzer:
  plugins:
    - custom_lint

custom_lint:
  enable_all_lint_rules: true
  rules:

```

**Expected behavior**
It should work

**Comment by emirchus** [comment]
Custom_lint.log:

```
[provider_parameters] 2023-02-13T09:23:25.524448 Plugin provider_parameters threw while analyzing /mnt/zekito/ottaa_project_flutter/lib/presentation/screens/link/ui/token_input.dart:
[provider_parameters] 2023-02-13T09:23:25.524448 Null check operator used on a null value
[provider_parameters] 2023-02-13T09:23:25.524448 #0      TypeChecker.isExactlyType (package:custom_lint_core/src/type_checker.dart:195:74)
[provider_parameters] 2023-02-13T09:23:25.524448 #1      WidgetRefInvocation.parse (package:riverpod_analyzer_utils/src/riverpod_ast.dart:60:32)
[provider_parameters] 2023-02-13T09:23:25.524448 #2      RiverpodRegistry._addWidgetRefVisitor.<anonymous closure> (package:riverpod_analyzer_utils/src/riverpod_visitor.dart:362:37)
[provider_parameters] 2023-02-13T09:23:25.524448 #3      _rootRunUnary (dart:async/zone.dart:1414:13)
[provider_parameters] 2023-02-13T09:23:25.524448 #4      _CustomZone.runUnary (dart:async/zone.dart:1307:19)
[provider_parameters] 2023-02-13T09:23:25.524448 #5      LinterVisitor._runSubscriptions (package:custom_lint_core/src/node_lint_visitor.dart:1034:27)
[provider_parameters] 2023-02-13T09:23:25.524448 #6      LinterVisitor.visitMethodInvocation (package:custom_lint_core/src/node_lint_visitor.dart:603:5)
[provider_parameters] 2023-02-13T09:23:25.524448 #7      MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8906:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #8      ExpressionFunctionBodyImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:4447:17)
[provider_parameters] 2023-02-13T09:23:25.524448 #9      LinterVisitor.visitExpressionFunctionBody (package:custom_lint_core/src/node_lint_visitor.dart:310:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #10     ExpressionFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:4439:15)
[provider_parameters] 2023-02-13T09:23:25.524448 #11     FunctionExpressionImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:6102:11)
[provider_parameters] 2023-02-13T09:23:25.524448 #12     LinterVisitor.visitFunctionExpression (package:custom_lint_core/src/node_lint_visitor.dart:418:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #13     FunctionExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:6091:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #14     NamedExpressionImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:9150:17)
[provider_parameters] 2023-02-13T09:23:25.524448 #15     LinterVisitor.visitNamedExpression (package:custom_lint_core/src/node_lint_visitor.dart:616:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #16     NamedExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:9140:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #17     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9445:20)
[provider_parameters] 2023-02-13T09:23:25.524448 #18     ArgumentListImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:401:16)
[provider_parameters] 2023-02-13T09:23:25.524448 #19     LinterVisitor.visitArgumentList (package:custom_lint_core/src/node_lint_visitor.dart:52:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #20     ArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:397:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #21     InstanceCreationExpressionImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:7569:19)
[provider_parameters] 2023-02-13T09:23:25.524448 #22     LinterVisitor.visitInstanceCreationExpression (package:custom_lint_core/src/node_lint_visitor.dart:508:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #23     InstanceCreationExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:7558:15)
[provider_parameters] 2023-02-13T09:23:25.524448 #24     ReturnStatementImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:11327:18)
[provider_parameters] 2023-02-13T09:23:25.524448 #25     LinterVisitor.visitReturnStatement (package:custom_lint_core/src/node_lint_visitor.dart:812:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #26     ReturnStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:11323:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #27     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9445:20)
[provider_parameters] 2023-02-13T09:23:25.524448 #28     BlockImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1333:17)
[provider_parameters] 2023-02-13T09:23:25.524448 #29     LinterVisitor.visitBlock (package:custom_lint_core/src/node_lint_visitor.dart:106:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #30     BlockImpl.accept (package:analyzer/src/dart/ast/ast.dart:1329:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #31     BlockFunctionBodyImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1284:12)
[provider_parameters] 2023-02-13T09:23:25.524448 #32     LinterVisitor.visitBlockFunctionBody (package:custom_lint_core/src/node_lint_visitor.dart:112:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #33     BlockFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:1276:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #34     MethodDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:8762:11)
[provider_parameters] 2023-02-13T09:23:25.524448 #35     LinterVisitor.visitMethodDeclaration (package:custom_lint_core/src/node_lint_visitor.dart:598:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #36     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8754:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #37     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9445:20)
[provider_parameters] 2023-02-13T09:23:25.524448 #38     ClassDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2090:13)
[provider_parameters] 2023-02-13T09:23:25.524448 #39     LinterVisitor.visitClassDeclaration (package:custom_lint_core/src/node_lint_visitor.dart:160:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #40     ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:2080:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #41     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9445:20)
[provider_parameters] 2023-02-13T09:23:25.524448 #42     CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2596:21)
[provider_parameters] 2023-02-13T09:23:25.524448 #43     LinterVisitor.visitCompilationUnit (package:custom_lint_core/src/node_lint_visitor.dart:184:10)
[provider_parameters] 2023-02-13T09:23:25.524448 #44     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2589:50)
[provider_parameters] 2023-02-13T09:23:25.524448 #45     DartLintRule.startUp.<anonymous closure> (package:custom_lint_core/src/lint_rule.dart:125:17)
[provider_parameters] 2023-02-13T09:23:25.524448 #46     _rootRun (dart:async/zone.dart:1398:13)
[provider_parameters] 2023-02-13T09:23:25.524448 #47     _CustomZone.run (dart:async/zone.dart:1300:19)
[provider_parameters] 2023-02-13T09:23:25.524448 #48     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1232:23)
[provider_parameters] 2023-02-13T09:23:25.524448 #49     runPostRunCallbacks (package:custom_lint_core/src/plugin_base.dart:14:19)
[provider_parameters] 2023-02-13T09:23:25.524448 #50     _ClientAnalyzerPlugin.analyzeFile.<anonymous closure> (package:custom_lint_builder/src/client.dart:772:7)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #51     _ClientAnalyzerPlugin._runOperation (package:custom_lint_builder/src/client.dart:815:14)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #52     _ClientAnalyzerPlugin.analyzeFile (package:custom_lint_builder/src/client.dart:740:5)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #53     ServerPlugin.analyzeFiles (package:analyzer_plugin/plugin/plugin.dart:119:7)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #54     ServerPlugin.afterNewContextCollection.<anonymous closure> (package:analyzer_plugin/plugin/plugin.dart:86:7)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #55     ServerPlugin._forAnalysisContexts (package:analyzer_plugin/plugin/plugin.dart:525:7)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #56     ServerPlugin.afterNewContextCollection (package:analyzer_plugin/plugin/plugin.dart:84:5)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #57     _ClientAnalyzerPlugin._runOperation (package:custom_lint_builder/src/client.dart:815:14)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #58     ServerPlugin.handleAnalysisSetContextRoots (package:analyzer_plugin/plugin/plugin.dart:273:5)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #59     _ClientAnalyzerPlugin._runOperation (package:custom_lint_builder/src/client.dart:815:14)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #60     ServerPlugin._getResponse (package:analyzer_plugin/plugin/plugin.dart:544:18)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448 #61     ServerPlugin._onRequest (package:analyzer_plugin/plugin/plugin.dart:611:18)
[provider_parameters] 2023-02-13T09:23:25.524448 <asynchronous suspension>
[provider_parameters] 2023-02-13T09:23:25.524448

```

**Comment by rrousselGit** [comment]
Could you share your Dart file?

**Comment by emirchus** [comment]
Yep
```dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import 'package:ottaa_project_flutter/application/providers/link_provider.dart';
import 'package:ottaa_project_flutter/application/router/app_routes.dart';
import 'package:ottaa_project_flutter/presentation/common/ui/loading_modal.dart';
import 'package:ottaa_project_flutter/presentation/screens/link/ui/token_input.dart';
import 'package:ottaa_ui_kit/widgets.dart';
import 'package:ottaa_project_flutter/application/common/extensions/translate_string.dart';

class OTPWidget extends ConsumerStatefulWidget {
  const OTPWidget({super.key});

  @override
  ConsumerState<ConsumerStatefulWidget> createState() => _OTPWidgetState();
}

class _OTPWidgetState extends ConsumerState<OTPWidget> {
  @override
  Widget build(BuildContext context) {
    final provider = ref.watch(linkProvider);
    final size = MediaQuery.of(context).size;

    return Form(
      key: provider.codeFormKey,
      child: SizedBox(
        height: 100,
        width: size.width * 0.8,
        child: Flex(
          direction: Axis.horizontal,
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: List.generate(7, (index) {
            if (index % 2 != 0) {
              return const Spacer();
            }

            final tokenId = index ~/ 2;

            return Expanded(
              flex: 3,
              child: TokenInput(
                tokenId: tokenId,
                controller: provider.controllers[tokenId],
                node: provider.focusNodes[tokenId],
                onChanged: (_, value) async {
                  provider.tokenChanged(tokenId, value);
                  bool isCode = provider.isValidCode();
                  if (isCode) {
                    bool isValid = false;
                    await LoadingModal.show(context, future: () async {
                      isValid = await provider.validateCode() == null;
                    });

                    if (!isValid) {
                      //TODO Emir check that this is OK
                      OTTAANotification.secondary(context, text: "profile.link.token.invalid".trl);
                      return;
                    }
                    context.push(AppRoutes.linkWaitScreen);
                  }
                },
              ),
            );
          }),
        ),
      ),
    );
  }
}
```

**Comment by rrousselGit** [comment]
Fixed by [65c0c0db96d725ac1e08e5c5d2536cd5c5e4fb15](https://github.com/invertase/dart_custom_lint/commit/65c0c0db96d725ac1e08e5c5d2536cd5c5e4fb15)
